support/testsupport/src/main/java/com/hotels/styx/support/dns/LocalNameServiceDescriptor.java

```diff
@@ -0,0 +1,58 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.support.dns;
+
+import sun.net.spi.nameservice.NameService;
+import sun.net.spi.nameservice.NameServiceDescriptor;
+
+import java.util.Optional;
+
+/*
```

Comment:
Maybe we can document this in the Javadoc? I was trying to find out what are the recommendations for Javadoc docs regarding concurrency but I haven't found anything in a very very quick search... We could also want to use annotations such as "@NotThreadSafe".

Reply:
Hmm... I assume you are referring to the `MockNameService` class?

I think the `LocalNameServiceDescriptor` itself is thread safe, but the consumers of the `MockNameService` should follow certain usage patterns.

I will add them in.